core/src/main/resources/jenkins/model/Jenkins/_restart.jelly

```diff
@@ -28,10 +28,17 @@ THE SOFTWARE.
 	<l:layout title="Restart Jenkins">
 		<st:include page="sidepanel.jelly" />
 		<l:main-panel>
-		  <form method="post" action="restart">
-		    ${%Are you sure about restarting Jenkins?}
-		    <f:submit value="${%Yes}" />
-		  </form>
+			<j:choose>
+				<j:when test="${app.lifecycle.canRestart()}">
+					<form method="post" action="restart">
+						${%Are you sure about restarting Jenkins?}
+						<f:submit value="${%Yes}" />
+					</form>
+				</j:when>
+				<j:otherwise>
+					${%Jenkins cannot restart itself as currently configured.}
```

Comment:
Minor comment.
I would move this phrase to the localization dictionary. It's too long for a key IMO. However, there's already a sentence ("Are you sure...") being used in such way, so probably there's no extra problems